src/rag/system.py

```diff
@@ -53,10 +49,11 @@ def __init__(self, models: dict[str, IModel | Embeddings]):
         self.query_generator = QueryGenerator(
             cast(IModel, cast(IModel, models[ModelType.GPT4O_MINI]))
         )
+        hana_conn = next(get_hana_connection()).connection
```

Comment:
What will happen if `get_hana_connection()` returns `None`? Would `None.connection` cause an exception here?

Reply:
I just noticed that I no longer need a generator so I will just refactor that func. Makes everything easier.